dist/gcc-compatible/Makefile

```diff
@@ -71,9 +70,13 @@ ifneq (,$(wildcard EverCrypt_BCrypt.h))
   SOURCES	+= evercrypt_bcrypt.c
 endif
 
+-include Makefile.config
```

Comment:
Ah, I see how you want to do this. Sure that makes sense.
But I think I'd generally (not in this PR) clean up this Makefile to get rid of this `if` block. In my mind (which is following the ng/gyp and ninja approach) the configuration should all happen *before* `make` is run, i.e. in `configure` at the moment. Then there's only one place where configuration happens and the Makefile is a simple set of linear rules.

Reply:
It would be nice. I think the reason I refrained from depending more on the configure script is that most people in everest are used to running "make" without running configure. But I think it's ok to be more stringent about it, or to have, in Makefile:

```
include Makefile.config

Makefile.config: configure
  ./$<
```

or something like taht

Reply:
which is to say, I would be very much in support of cleaning all of that up and relying more on the configure script everywhere!